src/reportSources/legacyFflogs/eventAdapter/prepullStatus.ts

```diff
@@ -0,0 +1,110 @@
+import {getDataBy, getDataArrayBy} from 'data'
+import ACTIONS from 'data/ACTIONS'
+import STATUSES, {Status} from 'data/STATUSES'
+import {Event, Events} from 'event'
+import _ from 'lodash'
+import {Actor} from 'report'
+import {AdapterStep} from './base'
+
+type StatusEvent = Events['statusApply'] | Events['statusRemove']
+type StatusKey = keyof typeof STATUSES
+
+// Statuses applied before the pull won't have an applyStatus event
+// Fabricate status applications so modules don't need to take this into account
+export class PrepullStatusAdapterStep extends AdapterStep {
+	private observedActions = new Map<Actor['id'], Set<number>>()
+	private observedStatuses = new Map<Actor['id'], Set<number>>()
+	private precastEvents: Event[] = []
+
+	postprocess(adaptedEvents: Event[]): Event[] {
+		for (const event of adaptedEvents) {
+			if (event.type !== 'statusApply' && event.type !== 'statusRemove') {
+				if (event.type === 'action') {
+					this.observeAction(event.action, event.source)
+				}
+				continue
+			}
+
+			const status = getDataBy(STATUSES, 'id', event.status)
```

Comment:
ngl i'm really pleased the event narrowed the way it did. i know it _should_, but it's nice to see it actually doing it in the wild